tests/config/test_email.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tempfile
+
+from synapse.util.stringutils import random_string
+
+from tests import unittest
+
+
+class EmailConfigTestCase(unittest.HomeserverTestCase):
+    def prepare(self, reactor, clock, hs):
+        self.hs = hs
+
+    def test_loading_missing_templates(self):
```

Comment:
What happens if the custom directory doesn't exist? Is that worth having a test for? Do things explode?

Reply:
I suppose we should probably raise a `ConfigError` in that case?

Reply:
Probably! I guess we only really need it there during initialization since we hold the templates in memory.

Reply:
We now raise a `ConfigError` if the given template directory does not exist.